pkg/kubelet/dockershim/exec.go

```diff
@@ -109,29 +109,42 @@ func (*NativeExecHandler) ExecInContainer(client libdocker.Interface, container
 		return err
 	}
 
+	var execTimeout <-chan time.Time
+	if timeout > 0 {
+		execTimeout = time.After(timeout)
+	} else {
+		// skip exec timeout if provided timeout is 0
+		execTimeout = make(chan time.Time, 1)
+	}
+
 	ticker := time.NewTicker(2 * time.Second)
 	defer ticker.Stop()
 	count := 0
 	for {
-		inspect, err2 := client.InspectExec(execObj.ID)
-		if err2 != nil {
-			return err2
-		}
-		if !inspect.Running {
-			if inspect.ExitCode != 0 {
-				err = &dockerExitError{inspect}
+		select {
+		case <-execTimeout:
+			return fmt.Errorf("timed out waiting for exec commands %v to finish for container %s", cmd, container.ID)
+		default:
```

Comment:
This needs to be `default` instead of `<-ticker.C`, otherwise we delay the initial InspectExec by 2 seconds.

Reply:
Done, thanks